### PR TITLE
(Pipeline Feature Branch) CMR-8380: add provider verification and ACL authorization to ingest o…

### PR DESCRIPTION
**Pull request by zimzoom**
…f generics, add relevant section to ingest permissions system int test

### CODE REVIEW

File: ingest-app/src/cmr/ingest/api/generic_documents.clj [hunk 1]
```diff
@@ -92,7 +92,10 @@
         provider-id (or (:provider params)
                         (:provider-id route-params))
         native-id (:native-id route-params)
-        ; TODO: Generic work - add token check
+        _ (lt-validation/validate-launchpad-token request-context)
+        _ (api-core/verify-provider-exists request-context provider-id)
+        _ (acl/verify-ingest-management-permission
+         request-context :update :provider-object provider-id)
```
Comment (jceaser) [inline ingest-app/src/cmr/ingest/api/generic_documents.clj:98]: is this how the editor spaced things?

File: system-int-test/src/cmr/system_int_test/utils/generic_util.clj [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   [cmr.transmit.config :as transmit-config]
+   [cmr.umm-spec.versioning :as umm-versioning]
+   [cmr.umm.echo10.echo10-collection :as c]
+   [cmr.umm.echo10.echo10-core :as echo10]
+   [cmr.umm.echo10.granule :as g])
+  (:import
+   [java.util UUID]))
+
+(def grid-good
+  {:MetadataSpecification {:URL "https://cdn.earthdata.nasa.gov/generic/grid/v0.0.1"
```
Comment (jceaser) [inline system-int-test/src/cmr/system_int_test/utils/generic_util.clj:33]: if you have the the schemas in system-int-test, you can replace this with the following:

`(def grid-good (-> "schemas/grid/v0.0.1/metadata.json"
                    (jio/resource)
                    (slurp)
                    (json/parse-string true)))`

File: system-int-test/src/cmr/system_int_test/utils/ingest_util.clj [hunk 3]
```diff
@@ -5,7 +5,7 @@
    [clojure.data :as d]
    [clojure.data.xml :as x]
    [clojure.string :as str]
-   [clojure.test :refer :all]
+   [clojure.test :refer :all]
```
Comment (jceaser) [inline system-int-test/src/cmr/system_int_test/utils/ingest_util.clj:8]: extra space

File: system-int-test/test/cmr/system_int_test/ingest/provider_ingest_permissions_test.clj [hunk 4]
```diff
@@ -298,7 +304,25 @@
             another-prov-admin-token
             "provider-admin-read-token can't ingest"
             provider-admin-read-token))
+     
+     ;; Generic Document Pipeline Prototype
```
Comment (jceaser) [inline system-int-test/test/cmr/system_int_test/ingest/provider_ingest_permissions_test.clj:308]: drop prototype

File: system-int-test/src/cmr/system_int_test/utils/ingest_util.clj [hunk 5]
```diff
@@ -918,7 +918,7 @@
      (dev-sys-util/reset)
      (set-ingest-umm-version-to-current)
      (when-not (empty? providers)
-      (setup-providers providers options))
+       (setup-providers providers options))
```
Comment (jceaser) [inline system-int-test/src/cmr/system_int_test/utils/ingest_util.clj:921]: could you say `(when (seq providers)) ...)` instead?

File: system-int-test/test/cmr/system_int_test/ingest/provider_ingest_permissions_test.clj [hunk 6]
```diff
@@ -298,7 +304,25 @@
...
+     (testing "ingest generic (grid) update permissions"
+       (are [token]
```
Comment (jceaser) [inline system-int-test/test/cmr/system_int_test/ingest/provider_ingest_permissions_test.clj:310]: why are you not using are3? I think that is the preferred standard way.

File: system-int-test/src/cmr/system_int_test/utils/ingest_util.clj (hunk 5 above)
Comment (zimzoom) [reply system-int-test/src/cmr/system_int_test/utils/ingest_util.clj:921]: good catch!

File: ingest-app/src/cmr/ingest/api/generic_documents.clj (hunk 1 above)
Comment (zimzoom) [reply ingest-app/src/cmr/ingest/api/generic_documents.clj:98]: we taled and decided to leave this for now.

**Review by eereiter — APPROVED** [review]

**Comment by eereiter** [comment]
Fix the extra space that Thomas stated and then you can squash and merge since Thomas is out for the week.

**Pull request merged**